pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -0,0 +1,428 @@
+		return reconcile.Result{}, err
+	}
+
+	var managerInternalTLSSecret certificatemanagement.CertificateInterface
+	if managementCluster != nil {
+		managerInternalTLSSecret, err = certificateManager.GetCertificate(r.client, render.ManagerInternalTLSSecretName, common.OperatorNamespace())
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("failed to retrieve / validate  %s", render.ManagerInternalTLSSecretName), err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	esgwCertificate, err := certificateManager.GetCertificate(r.client, relasticsearch.PublicCertSecret, common.OperatorNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, fmt.Sprintf("Failed to retrieve / validate  %s", relasticsearch.PublicCertSecret), err, reqLogger)
+		return reconcile.Result{}, err
+	} else if esgwCertificate == nil {
+		log.Info("Elasticsearch gateway certificate is not available yet, waiting until they become available")
+		r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch gateway certificate are not available yet, waiting until they become available", nil, reqLogger)
+		return reconcile.Result{}, nil
+	}
+	trustedBundle := certificateManager.CreateTrustedBundle(managerInternalTLSSecret, esgwCertificate)
+
+	certificateManager.AddToStatusManager(r.status, render.PolicyRecommendationNamespace)
+
+	// Create a component handler to manage the rendered component.
+	handler := utils.NewComponentHandler(log, r.client, r.scheme, instance)
+
+	reqLogger.V(3).Info("rendering components")
+	hasNoLicense := !utils.IsFeatureActive(license, common.PolicyRecommendationFeature)
+	openshift := r.provider == operatorv1.ProviderOpenShift
+	policyRecommendationCfg := &render.PolicyRecommendationConfiguration{
+		ClusterDomain:   r.clusterDomain,
+		ESClusterConfig: esClusterConfig,
+		ESSecrets:       esSecrets,
+		Installation:    network,
+		PullSecrets:     pullSecrets,
+		TrustedBundle:   trustedBundle,
+		Openshift:       openshift,
+		UsePSP:          r.usePSP,
+	}
+
+	// Render the desired objects from the CRD and create or update them.
+	component := render.PolicyRecommendation(policyRecommendationCfg)
+
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, component); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	components := []render.Component{
+		component,
+		rcertificatemanagement.CertificateManagement(&rcertificatemanagement.Config{
+			Namespace:       render.PolicyRecommendationNamespace,
+			ServiceAccounts: []string{render.PolicyRecommendationName},
+			KeyPairOptions: []rcertificatemanagement.KeyPairOption{
+				rcertificatemanagement.NewKeyPairOption(policyRecommendationCfg.PolicyRecommendationServerCertSecret, true, true),
+			},
+			TrustedBundle: trustedBundle,
+		}),
+	}
+
+	if hasNoLicense {
```

Comment:
Nit, but it's much nicer to perform checks like these right after calculating the variable. Makes the code easier to read and harder to introduce accidental bugs.

```go
hasNoLicense := !utils.IsFeatureActive(license, common.PolicyRecommendationFeature)
```

can be moved down here. 

Since it's only used in one place, you could even do:

```go
if !utils.IsFeatureActive(license, common.PolicyRecommendationFeature) { . . . }
```